<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>qSlicerCropVolumeModuleWidget</class>
 <widget class="qSlicerWidget" name="qSlicerCropVolumeModuleWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>682</width>
    <height>358</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Crop Volume</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="ctkCollapsibleButton" name="ParametersCollapsibleButton" native="true">
     <property name="text" stdset="0">
      <string>Parameters</string>
     </property>
     <layout class="QFormLayout" name="parametersFormLayout">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::FieldsStayAtSizeHint</enum>
      </property>
      <property name="verticalSpacing">
       <number>12</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="InputVolumeLabel">
        <property name="text">
         <string>Input volume:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="qMRMLNodeComboBox" name="InputVolumeComboBox" native="true">
        <property name="nodeTypes" stdset="0">
         <stringlist>
          <string>vtkMRMLScalarVolumeNode</string>
         </stringlist>
        </property>
        <property name="addEnabled" stdset="0">
         <bool>false</bool>
        </property>
        <property name="renameEnabled" stdset="0">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="InputROILabel">
        <property name="text">
         <string>Input ROI:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="qMRMLNodeComboBox" name="InputROIComboBox" native="true">
        <property name="nodeTypes" stdset="0">
         <stringlist>
          <string>vtkMRMLAnnotationROINode</string>
         </stringlist>
        </property>
        <property name="editEnabled" stdset="0">
         <bool>true</bool>
        </property>
        <property name="renameEnabled" stdset="0">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="ROIVisibilityLabel">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>ROI visibility:</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QWidget" name="EmptyWidget" native="true"/>
      </item>
      <item row="7" column="0" colspan="2">
       <widget class="ctkCollapsibleGroupBox" name="InterpolationOptionsGroupBox">
        <property name="title">
         <string>Interpolation options</string>
        </property>
        <layout class="QFormLayout" name="formLayout_2">
         <property name="fieldGrowthPolicy">
          <enum>QFormLayout::ExpandingFieldsGrow</enum>
         </property>
         <property name="margin">
          <number>0</number>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="IsotropicOutputVoxelLabel">
           <property name="text">
            <string>Isotropic output voxel:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QCheckBox" name="IsotropicCheckbox">
           <property name="text">
            <string/>
           </property>
           <property name="tristate">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="InputSpacingScalingConstantLabel">
           <property name="toolTip">
            <string>In not equal to 1, this will result in upsampling (&lt;1) or downlsampling (&gt;1) relative to the voxel spacing of the input volume.</string>
           </property>
           <property name="text">
            <string>Input spacing scaling constant:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="ctkSpinBox" name="SpacingScalingSpinBox">
           <property name="toolTip">
            <string>The voxel spacing in the output volume will be scaled by this value.</string>
           </property>
           <property name="value">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="InterpolatorLabel">
           <property name="text">
            <string>Interpolator:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QWidget" name="InterpolatorWidget" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <property name="margin">
             <number>0</number>
            </property>
            <item>
             <widget class="QRadioButton" name="NNRadioButton">
              <property name="text">
               <string>Nearest Neighbor</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="LinearRadioButton">
              <property name="text">
               <string>Linear</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="WSRadioButton">
              <property name="text">
               <string>WindowedSinc</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="BSRadioButton">
              <property name="text">
               <string>B-spline</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="TechniqueLabel">
        <property name="text">
         <string>Technique:</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QToolButton" name="VisibilityButton">
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset>
          <normaloff>:/Icons/VisibleOff.png</normaloff>
          <normalon>:/Icons/VisibleOn.png</normalon>:/Icons/VisibleOff.png</iconset>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <layout class="QHBoxLayout" name="TechniqueRadioButtonsHorizontalLayout">
        <property name="spacing">
         <number>12</number>
        </property>
        <item>
         <widget class="QRadioButton" name="InterpolationModeRadioButton">
          <property name="text">
           <string>Interpolated cropping</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
          <attribute name="buttonGroup">
           <string>TechniqueButtonGroup</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="VoxelBasedModeRadioButton">
          <property name="text">
           <string>Voxel based cropping</string>
          </property>
          <attribute name="buttonGroup">
           <string>TechniqueButtonGroup</string>
          </attribute>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="CropButton">
     <property name="text">
      <string>Crop!</string>
     </property>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qSlicerWidget</class>
   <extends>QWidget</extends>
   <header>qSlicerWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkCollapsibleButton</class>
   <extends>QWidget</extends>
   <header>ctkCollapsibleButton.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkCollapsibleGroupBox</class>
   <extends>QGroupBox</extends>
   <header>ctkCollapsibleGroupBox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkSpinBox</class>
   <extends>QWidget</extends>
   <header>ctkSpinBox.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../../../Libs/MRML/Widgets/Resources/qMRMLWidgets.qrc"/>
  <include location="../qSlicerCropVolumeModule.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>qSlicerCropVolumeModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>InputVolumeComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>245</x>
     <y>4</y>
    </hint>
    <hint type="destinationlabel">
     <x>376</x>
     <y>111</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerCropVolumeModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>InputROIComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>352</x>
     <y>7</y>
    </hint>
    <hint type="destinationlabel">
     <x>483</x>
     <y>148</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>InterpolationModeRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>InterpolationOptionsGroupBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>254</x>
     <y>83</y>
    </hint>
    <hint type="destinationlabel">
     <x>340</x>
     <y>161</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>InterpolationModeRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>InterpolationOptionsGroupBox</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>77</y>
    </hint>
    <hint type="destinationlabel">
     <x>340</x>
     <y>221</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <buttongroups>
  <buttongroup name="TechniqueButtonGroup"/>
 </buttongroups>
</ui>
